c7n/resources/ebs.py

```diff
@@ -289,9 +289,10 @@ def _pull_asg_snapshots(self):
         tmpl_mgr = self.manager.get_resource_manager('launch-template-version')
         for tversion in tmpl_mgr.get_resources(
                 list(tmpl_mgr.get_asg_templates(asgs).keys())):
-            for bd in tversion['LaunchTemplateData']['BlockDeviceMappings']:
-                if 'Ebs' in bd and 'SnapshotId' in bd['Ebs']:
-                    snap_ids.add(bd['Ebs']['SnapshotId'])
+            if 'BlockDeviceMappings' in tversion['LaunchTemplateData']:
+                for bd in tversion['LaunchTemplateData']['BlockDeviceMappings']:
```

Comment:
preference would be to use `.get('BlockDeviceMappings', ()):` which avoids the extra indent/conditional

Reply:
Fixed it. Thank you for the review.